tests/drivers/i2c/i2c_sensor/testcase.yaml

```diff
@@ -0,0 +1,7 @@
+common:
+  tags: drivers i2c
+  depends_on: i2c
+  harness: ztest
+tests:
+  drivers.i2c.controller_to_sensor:
+    filter: dt_alias_exists("sensor-bme688")
```

Comment:
How will this test be built in CI?

Reply:
Until better solution is available we will use Twister's '--alt-config-root' to override the testcase.yaml (for the internal shield).

Reply:
That doesn't answer the question of how this will be built in upstream CI.

Why not go with a solution just depending on a given compatible being enabled?

Reply:
Could you provide an example of such syntax ?

Reply:
Note: boards/nordic/thingy53/thingy53_nrf5340_common.dtsi has this sensor, so this can be enabled upstream. You could filter based on `dt_compat_enabled`, which doesn't require any aliases to be present.